app/models/staff_availability.rb

```diff
@@ -2,12 +2,16 @@ class StaffAvailability < ApplicationRecord
   belongs_to :user, optional: true
   belongs_to :time_period
 
-  validates :day, presence: true
-  validates :start_time, presence: true
-  validates :end_time, presence: true
   validates :time_period_id, presence: true
 
+  validates :start_time, :end_time, :day, presence: true, if: :recurring?
+  validates :start_datetime, :end_datetime, presence: true, unless: :recurring?
+
   def week_day
     Date::DAYNAMES[day.to_i]
   end
+
+  def recurring?
```

Comment:
Double check in rails docs but I believe this already exists for all the booleans automatically (might be wrong tho)